lua/telescope/previewers.lua

```diff
@@ -401,6 +405,12 @@ previewers.help = defaulter(function(_)
         vim.cmd "norm! gg"
         vim.fn.search(search_query, "W")
         vim.cmd "norm  zt"
+
+        if status.preview_search_hl_id then
+          vim.fn.matchdelete(status.preview_search_hl_id)
+          status.preview_search_hl_id = nil
```

Comment:
we don't want to use `status` here, the `previewer` has a `state` field, that you can `setup` and `teardown` exactly for this kind of thing (so you can clear the highlight when we're done w/ the previewer as well.

Reply:
Ping on this, basically only thing stopping from merging.